elastic/kibana
# [Reporting/Visualization] Migrate Visualize to V2 reporting

## Summary

Updates Visualize to use the new v2 report (powered by locators) that ensures all state can be migrated and as a bonus removes the requirement to first save changes.

New export types were introduced in https://github.com/elastic/kibana/pull/101048.


## How to test

1. Go "Visualize Library"
2. Create a new visualization, one that uses "app/visualize". This includes "Custom visualiztions" powered by Vega. Installing a sample data set should get you some of these
3. Edit the visualization in some way, press "Update"
4. Generate a PDF via "Share" menu, this should now be clickable without first needing to Save the visualization.
5. Do steps 3 & 4 but for an existing visualisation
6. In both cases you should get a report (PDF/PNG) with the latest changes you updated to

## Release note

You can now generate a PDF or PNG report of an unsaved visualization.

<img width="385" alt="Screenshot 2021-08-26 at 13 35 15" src="https://user-images.githubusercontent.com/8155004/130955909-9cfb8625-6ffa-4e60-80fb-454440afafe1.png">

### Checklist
